python/lsst/dax/imgserv/cli.py

```diff
@@ -0,0 +1,162 @@
+        log.configure(os.path.join(config_dir, "log.properties"))
+        self._in_req = in_req
+        self._out_dir = out_dir
+        self._dispatcher = Dispatcher(os.path.join(ROOT, "config"))
+
+    def _fix_dataroot_path(self):
+        # this is needed to accommodate cur dir being different
+        # when running inside container image
+        img_ds = str(self._config["DAX_IMG_DS"])
+        img_dr = str(self._config["DAX_IMG_DR"])
+        self._config["DAX_IMG_DS"] = img_ds.replace("./tests", "./")
+        self._config["DAX_IMG_DR"] = img_dr.replace("./tests", "./")
+
+    def process_request(self):
+        errors, req = self._parse_req()
+        w13db = self._get_ds(req)
+        if w13db is not None:
+            img_getter = image_open(w13db, self._config)
+            # keep this specific instance of img_getter
+            Image.imagegetter = img_getter
+            result = self.dispatch(req)
+            name = req["name"]
+            fp_image = self._save_result(result, name) 
+            if self._check_result(req, fp_image):
+                print(name+": PASS")
+            else:
+                print(name+": FAIL")
+        else:
+            raise Exception("ImageServCLI", "w13db should NOT be None")
+
+    def dispatch(self, req):
+        # direct the request to the best fit image API on parameters
+        api, params = self._dispatcher.find_api(req)
+        # call the API with the params
+        if api is not None:
+            result = api(params)
+            return result
+        
+    def _save_result(self, image, fn_prefix="image_out"):
+        # save image object into fits file in output directory
+        fn = os.path.join(self._out_dir, fn_prefix+".fits")
+        if image is None:
+            with open(fn, "w") as f:
+                f.write("Image Not Found")
+                f.close()
+        else:
+            image.writeFits(fn) 
+        return fn
+
+    def _check_result(self, req, image_fp):
+        check_code = req["check"]["sha256"]
+        with open(image_fp, "rb") as f:
+            image_data = f.read()
+            f.close()
+        h = Hasher.hash(image_data).hexdigest()
+        return True if h == check_code else False
+
+    def _get_ds(self, req):
+        # determine and get the image datasource
+        image_type = req["get"]["api_key"][0]
+        if image_type == "raw":
+            return W13RawDb
+        elif image_type == "calexp":
+            return W13CalexpDb
+        elif image_type == "deepcoadd":
+            return W13DeepCoaddDb
+        else:
+            return None
+
+    def _get_req_data(self):
+        with open(self._in_req, "r") as req_file:
+            data = req_file.read()
+            req_file.close()
+        return data
+            
+    def _parse_req(self):
+        req_data = self._get_req_data()
+        # read and parse the request (JSON)
+        req = json.loads(req_data) 
+        # ToDo: validate schema here
```

Comment:
This looks important enough that a ticket should be created and its id added to this todo.